### PR TITLE
Revert "Deploy kubemacpool-mac-controller-manager with StatefulSet (#…

### PR DESCRIPTION
**Pull request by RamLavi**
…112)"

This reverts commit 33679fcc78d925f7afc9e0ac60003b0a34017de3.

```release-note
NONE
```

### CODE REVIEW

File: tests/tests.go [hunk 1]
```diff
@@ -128,9 +128,9 @@ func addNetworksToPod(pod *corev1.Pod, networks []map[string]string) {
 	}
 }
 
-func findPodByUid(pods *corev1.PodList, podToFind corev1.Pod) *corev1.Pod {
```
Comment (ormergi) [inline tests/tests.go:131]: We can keep this change, Its more accurate when looking for a pod.

Comment (RamLavi) [reply tests/tests.go:131]: yes, but let's insert it in a different PR, as it's not solely related to statefulSet, and I want to remove the commit as a whole

**Pull request merged**